python/examples/dipole_in_vacuum_1D.py

```diff
@@ -0,0 +1,354 @@
+            inverse_rotation_matrix = np.transpose(rotation_matrix)
+
+            farfields = {}
+            for component in farfield_components:
+                farfields[component] = np.zeros(
+                    (NUM_POLAR, NUM_AZIMUTHAL),
+                    dtype=np.complex128
+                )
+
+            for i, polar in enumerate(polar_rad):
+                for j, azimuthal in enumerate(azimuthal_rad):
+                    rx, ry, rz = spherical_to_cartesian(polar, azimuthal)
+                    r_rotated = rotation_matrix @ np.transpose(np.array([rx, ry]))
+                    # Note: r_rotated[1] (ry) is not used because k_rotated[1]=0 (ky).                       
+                    phase = np.exp(1j * 2 * np.pi * (k_rotated[0] * r_rotated[0] + kz * rz))
+
+                    # Obtain the farfields for each type of sheet current                                    
+                    # and keep a running sum.                                                                
+                    for current_component, current_amplitude in zip(
+                        current_components, current_amplitudes
+                    ):
+                        if abs(current_amplitude) == 0:
+                            continue
+
+                        farfield_amplitudes_pol = farfield_amplitudes(
+                            k_rotated[0],
+                            kz,
+                            r_rotated[0],
+                            rz,
+                            current_amplitude,
+                            current_component,
+                        )
+
+                        farfield_pol = np.array(farfield_amplitudes_pol) * phase
+
+                        if (
+                            current_component.name == "Jx" or
+                            current_component.name == "Ky"
+                        ):
+                            # P polarization                                                                 
+                            farfields["Ex"][i, j] += farfield_pol[0]
+                            farfields["Hy"][i, j] += farfield_pol[1]
+                            farfields["Ez"][i, j] += farfield_pol[2]
+                        elif (
+                            current_component.name == "Jy" or
+                            current_component.name == "Kx"
+                        ):
+                            # S polarization                                                                 
+                            farfields["Hx"][i, j] += farfield_pol[0]
+                            farfields["Ey"][i, j] += farfield_pol[1]
+                            farfields["Hz"][i, j] += farfield_pol[2]
+
+            for i in range(NUM_POLAR):
+                for j in range(NUM_AZIMUTHAL):
+                    # Map the rotated point to the closest azimuthal angle.                                  
+                    azimuthal_index = (j + azimuthal_index_shift) % NUM_AZIMUTHAL
```

Comment:
shifting the index seems wrong.   the point index hasn't changed in this process, you only rotated the fields and r.